staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -415,30 +415,26 @@ func (c *AvailableConditionController) sync(key string) error {
 
 				results <- nil
 			}()
-		}
 
-		var lastError error
-		for i := 0; i < attempts; i++ {
-			lastError = <-results
-			// if we had at least one success, we are successful overall and we can return now
+			lastError := <-results
```

Comment:
this looks to me like instead of running five concurrent requests and then checking the results, this launches a single go routine and immediately blocks waiting for the result.  If I'm right, this would defeat the intent  of running five parallel requests.

Reply:
@deads2k what's the intent of running five parallel requests ?